modules/shortcodes/facebook.php

```diff
@@ -40,6 +40,15 @@ function jetpack_facebook_embed_handler( $matches, $attr, $url ) {
 		return $embed . '<script src="//connect.facebook.net/en_US/all.js#xfbml=1"></script>';
 	} else {
 		wp_enqueue_script( 'jetpack-facebook-embed', plugins_url( 'js/facebook.js', __FILE__ ), array( 'jquery' ), null, true );
+		/** This filter is documented in modules/sharedaddy/sharing-sources.php */
+		$fb_app_id = absint( apply_filters( 'jetpack_sharing_facebook_app_id', '249643311490' ) );
+		wp_localize_script(
+			'jetpack-facebook-embed',
+			'jpfbembed',
+			array(
+				'appid' => $fb_app_id
```

Comment:
Not sure if Facebook will barf on an empty (non-numeric) app ID, but I suppose that's partly on the developer to ensure a valid ID is passed.